sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
Sorry for the late reply, but I am wondering if the ORC  / Parquet has the similar bug. 
https://github.com/apache/spark/pull/8023/files#diff-7c81d57959cd291f5297e4585fd7cdc3R273

Hopefully #8023 can fix this.